doc/source/walkthrough.rst

```diff
@@ -215,131 +381,278 @@ Object IDs can be created in multiple ways.
-<actors.html>`__ for more details.)
+    Actors extend the Ray API from functions (tasks) to classes. The ``Ray.createActor``
+    is used to create actors from normal Java classes. An actor is essentially a stateful worker.
+    Unlike Python, multiple Java actors may share one JVM process, in order to reduce JVM's memory
+    overhead. But this is transparent to normal users.
 
-.. code-block:: python
+    .. code-block:: java
 
-  @ray.remote(num_cpus=2, num_gpus=0.5)
-  class Actor(object):
-      pass
+      // A normal Java class.
+      public class Counter {
 
-We can interact with the actor by calling its methods with the ``.remote``
-operator. We can then call ``ray.get`` on the object ID to retrieve the actual
-value.
+        private int value = 0;
 
-.. code-block:: python
+        public int increment() {
+          this.value += 1;
+          return this.value;
+        }
+      }
 
-  obj_id = a1.increment.remote()
-  ray.get(obj_id) == 1
+      // Create an actor from this class.
+      // `Ray.createActor` takes a factory method that can produce
+      // a `Counter` object. Here, we pass `Counter`'s constructor
+      // as the argument.
+      RayActor<Counter> actor = Ray.createActor(Counter::new);
+
+    You can specify resource requirements in Actors too (see the `Actors section
+    <actors.html>`__ for more details).
+
+    .. code-block:: java
+
+      Map<String, Double> resources = ImmutableMap.of("CPU", 2, "GPU", 0.5);
+      ActorCreationOptions options = new ActorCreationOptions.Builder()
+          .setResources(resources).createActorCreationOptions();
+      Ray.createActor(Counter::new, options);
+
+    We can interact with the actor by calling its methods with the ``RayActor::call``
+    method, which also returns a ``RayObject``.
+
+    .. code-block:: java
+
+      RayObject<Integer> result = actor.call(Counter::increment);
+      Assert.assertEqual(result.get(), 1);
 
 
 Methods called on different actors can execute in parallel, and methods called on the same actor are executed serially in the order that they are called. Methods on the same actor will share state with one another, as shown below.
 
-.. code-block:: python
+.. tabs::
+
+  .. group-tab:: Python
+
+    .. code-block:: python
+
+      # Create ten Counter actors.
+      counters = [Counter.remote() for _ in range(10)]
+
+      # Increment each Counter once and get the results. These tasks all happen in
+      # parallel.
+      results = ray.get([c.increment.remote() for c in counters])
+      print(results)  # prints [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+
+      # Increment the first Counter five times. These tasks are executed serially
+      # and share state.
+      results = ray.get([counters[0].increment.remote() for _ in range(5)])
+      print(results)  # prints [2, 3, 4, 5, 6]
+
+  .. group-tab:: Java
 
-  # Create ten Counter actors.
-  counters = [Counter.remote() for _ in range(10)]
+    .. code-block:: java
```

Comment:
FYI, you can use `.. code-tab:: java` instead of `.. group-tab:: Java` and `.. code-block:: java`. See https://github.com/djungelorm/sphinx-tabs#code-tabs.